### PR TITLE
treasure chest and coffer rewrite

### PR DESCRIPTION
**Pull request by wrenffxi**
[CORE]
removed treasure_spawn_points.sql: points are now in lua
removed UpdateTreasureSpawnPoint from core: movement is now in lua

[LUA]
tabled treasure data
moved code from chest and coffer luas into treasure.lua
standardized chest and coffer NPC names in IDs.lua
removed chest-related text IDs from IDs.lua: use offsets from the first instead
weakness prevents you from picking locks, but does not spawn a mimic
failing a lockpick will either do nothing, give you weakness, or spawn a mimic (coffer only)
fixed a few incorrectly categorized mob and npc in IDs.lua
"just an illusion" cooldown is zone-wide rather than per-player

can clean up old char vars via
```delete FROM char_vars WHERE varname rlike '\[[0-9]*\]Treasure_C(hest|offer)'```

### CODE REVIEW

File: scripts/globals/treasure.lua [hunk 1]
```diff
@@ -1,493 +1,1540 @@
...
-        return {
-            dsp.job.BST,    BORGHERTZ_S_WILD_HANDS,         12517, -- Beast Helm
-            dsp.job.PLD,    BORGHERTZ_S_STALWART_HANDS,     12515, -- Gallant Coronet
-            dsp.job.WHM,    BORGHERTZ_S_HEALING_HANDS,      13855, -- Healer's Cap
-            dsp.job.MNK,    BORGHERTZ_S_STRIKING_HANDS,     12512, -- Temple Crown
-            dsp.job.RDM,    BORGHERTZ_S_VERMILLION_HANDS,   14218, -- Warlock's Tights
-        };
-    elseif (zone == 205) then -- Ifrit's Cauldron
-        return {
-
```
Comment (nfaletra) [inline scripts/globals/treasure.lua]: I blame Zynjec for making me panic.

`
local treasureMinLvl = info.treasureLvl - 10
if mJob ~= dsp.job.THF or mLvl < treasureMinLvl then
`
No need to store treasureMinLvl per zone if it's always 10 levels below the treasureLvl

Comment (nfaletra) [reply scripts/globals/treasure.lua]: github is fucking drunk

Comment (zynjec) [reply scripts/globals/treasure.lua]: Don't blame the messenger, I thought you just hated DRG.

**Comment by wrenffxi** [comment]
Removed treasureMinLvl.  Also, the "just an illusion" cooldown is zone-wide rather than a per-player.

**Comment by zynjec** [comment]
It looks like the `suffix` stuff can probably be gutted now with moving it to a local var on NPC.

Also, are we sure illusions are zone wide and not for player? The only real information suggesting that is the one paragraph on the wiki page about chests, but from the official patch notes, it sure sounds like it's per-player:

```
The respawn time for Treasure Coffers and Treasure Chests has been shortened, and the chances of obtaining artifacts, maps, and key items from them have been increased. Furthermore, after obtaining items or gil from a coffer or chest, players will temporarily become unable to receive items or gil from other coffers or chests.
```
http://www.playonline.com/pcd/verup/ff11us/detail/3893/detail.html

**Comment by wrenffxi** [comment]
here are the other references I found to it being zone-wide
[FFXIAH](https://www.ffxiah.com/forum/topic/1430/coffer-is-an-illusion#15220)
[BGWiki](https://www.bluegartr.com/threads/71745-Chest-Coffer-Fatigue-(Illusionary-Chest-BS)?p=2575271&viewfull=1#post2575271)

but i'm also finding people saying it's player-specific, so may need retail testing to vreify.

**Comment by wrenffxi** [comment]
rebased and removed unnecessary "suffix" property

**Comment by zynjec** [comment]
I did more research and I'm okay with illusions being zone wide unless someone can come forward with information to the contrary.

I'll leave this open for a bit to see if anyone else wants to chime in.

**Pull request merged**